src/Kinepolis/KinepolisDateParser.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Kinepolis;
+
+use CultuurNet\UDB3\Model\ValueObject\Calendar\BookingAvailability;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\BookingAvailabilityType;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\DateRange;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\Status;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\StatusType;
+use CultuurNet\UDB3\Model\ValueObject\Calendar\SubEvent;
+
+final class KinepolisDateParser implements DateParser
+{
+    // This class converts the Json received from the external MovieAPI,
+    // which has a hierarchy that is divided as follows:
+    // - movie (e.g., "Het smelt")
+    // - date (e.g., 2024-04-09)
+    // - screeningTime (e.g. 19:45:00)
+    // - version (2D or 3d)
+    // into a hierarchy that is better suited to convert them to UiTDatabank events by changing it to a
+    // - LocationId
+    // - version
+    // - DateTimeList
+    // hierarchy
+    private array $timeTableList;
+
+    public function processDates(array $dates, int $length): array
+    {
+        $this->timeTableList = [];
+
+        foreach ($dates as $day => $timeList) {
+            $this->processDay($day, $timeList, $length);
+        }
+
+        return $this->timeTableList;
+    }
+
+    private function processDay(string $day, array $timeList, int $length): void
+    {
+        foreach ($timeList as $info) {
+            $format = $this->getFormat($info['format']);
+            $from = $this->getFromTime($day, $info['time']);
+            $to = $this->getToTime($from, $length);
+            $this->timeTableList[$info['tid']][$format][] = new SubEvent(
```

Comment:
Do we need to stick with this format or could we also make it a typed object like `ParsedMovie`?

Reply:
I might be to hard to include it in the first release, but using using a typed Object like `ParsedMovie` could be an optimalization in the future.